src/crimson/admin/osd_admin.cc

```diff
@@ -208,14 +208,55 @@ class SeastarMetricsHook : public AdminSocketHook {
    for (const auto& [full_name, metric_family]: seastar::scollectd::get_value_map()) {
      for (const auto& [labels, metric] : metric_family) {
        if (metric && metric->is_enabled()) {
-         auto& metric_function = metric->get_function();
-         f->dump_float(metric->get_id().full_name(), metric_function().d());
+	 dump_metric_value(*metric);
        }
      }
    }
    f->close_section();
    return seastar::make_ready_future<tell_result_t>(std::move(f));
  }
+private:
+  using registered_metric = seastar::metrics::impl::registered_metric;
+  using data_type = seastar::metrics::impl::data_type;
+
+  void dump_metric_value(const registered_metric& metric)
+  {
+    switch (auto v = metric(); v.type()) {
+    case data_type::GAUGE:
+      f->dump_float(full_name, v.d());
```

Comment:
Oops. Looks like make check is failing because `full_name` needs to be metric.get_id().full_name() and `std::unique_ptr<Formatter> f` was not passed to the function

Reply:
@AmnonHanuhov fixed and repushed.